src/librustc_mir/transform/const_prop.rs

```diff
@@ -89,6 +89,7 @@ impl<'tcx> MirPass<'tcx> for ConstProp {
                 Default::default(),
                 tcx.def_span(source.def_id()),
                 Default::default(),
+                body.generator_kind, // FIXME: should this rather just be None since it is a dummy?
```

Comment:
cc @oli-obk @wesleywiser

Reply:
It's unproblematic to keep around, so I'd rather keep it around instead of suddenly wondering why we get weird error messages

Reply:
Ok will keep it as is.